spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -119,6 +124,31 @@ public Blob getGoogleStorageObject() throws IOException {
 		return this.storage.get(getBlobId());
 	}
 
+	/**
+	 * Gets a signed URL to an object if it exists.
+	 * @param timeUnit the time unit used to determine how long the URL is valid.
+	 * @param timePeriods the number of periods to determine how long the URL is valid.
+	 * @return the URL if the object exists, and null if it does not.
+	 * @throws IOException
+	 */
+	public URL getSignedUrl(TimeUnit timeUnit, int timePeriods) throws IOException {
+		Blob blob = this.getGoogleStorageObject();
+		if (blob == null) {
+			return null;
+		}
+		String privateKeyPath = System.getenv(
```

Comment:
Can we use the auto-configured `CredentialsProvider` to get the private key instead? @joaoandremartins

Reply:
Doesn't look like there's a way to get a private key from com.google.auth.Credentials. I'm not a security guy, but perhaps it's intentional that the only copy of the private key you're supposed to have is from the JSON file you download just once when creating the service account in the UI.

Reply:
What about the `GcpStorageProperties.credentials.location`? We already have a mechanism for specifying the private key file. Why use the environment variable instead?

Reply:
See my other comment. The prop you're thinking of is actually only in the starter projects, and infact the larger repo-level docs here specify using the env var. The env var is also standard across any GCP code, so I don't feel it's out of place (it's also mentioned to be set in this repo's docs anyway).

Reply:
Actually I could make this path a member that we set in the constructor (like the auto-create setting last time), and that it would just default to GOOGLE_APPLICATION_CREDENTIALS. Still, we would have this env var involved, but we could at least use the path from the starter instead if it was provided.